config/dependency.go

```diff
@@ -338,8 +338,14 @@ func getTerragruntOutputIfAppliedElseConfiguredDefault(dependencyConfig Dependen
 			case NoMerge:
 				return outputVal, nil
 			case ShallowMerge:
+				if dependencyConfig.MockOutputs == nil {
+					return outputVal, nil
+				}
```

Comment:
NIT: Should this check be above, around line 335? So we only enter this if statement block and switch statement if the `MockOutputs` isn't `nil`?

Reply:
Yes, simplified verification